configure.ac

```diff
@@ -1444,13 +1444,28 @@ if test -n "$LIBAIO"; then
 fi
 
 BUILD_LIBURING=no
-AC_CHECK_HEADERS([liburing.h])
-AC_CHECK_LIB([uring],[io_uring_queue_init],[LIBURING="-luring"])
+case $host_os in
+    linux*)
+        AC_ARG_ENABLE([linux_io_uring],
+                      AC_HELP_STRING([--disable-linux-io_uring],
+                      [Disable io-uring in POSIX xlator.]))
+
+        # Don't build on our jenkins CentOS 7 machines. We can remove this
+        # centOS7 variable when we start using CentOS Stream where liburing is available.
+        if test "x$ac_cv_file__etc_redhat_release" = "xyes"; then
+            if grep "release 7" /etc/centos-release; then
```

Comment:
If this is enabled by default, the glusterfs.spec can be used to pass `--disable-linux-io_uring` when it is building on CentOS 7. That way, there is not need for this hack in the configure script.